hivelytracker: init at unstable-2020-02-10

fgaz opened the pull request:
###### Motivation for this change

#81815 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@OPNA2608 here we go :)

fgaz committed:
hivelytracker: init at unstable-2020-02-10

fgaz commented on pkgs/applications/audio/hivelytracker/default.nix:
```diff
@@ -0,0 +1,74 @@
+  };
+
+  # https://github.com/pete-gordon/hivelytracker/pull/42
+  patches = [(fetchpatch {
```
Well, they merged it fast :D removing this now...

OPNA2608 commented on pkgs/applications/audio/hivelytracker/default.nix:
```diff
@@ -0,0 +1,76 @@
+
+  makeFlags = [ "-f Makefile.linux" "PREFIX=$(out)" ];
+
+  NIX_CFLAGS_COMPILE =
+    "-I${SDL}/include/SDL \
+     -I${SDL_image}/include/SDL \
+     -I${SDL_ttf}/include/SDL \
+     -I${gtk2.dev}/include/gtk-2.0 \
+     -I${glib.dev}/include/glib-2.0";
```
I think this'd be a little nicer, assuming the Darwin version doesn't use GTK2.

```suggestion
  NIX_CFLAGS_COMPILE = stdenv.lib.concatStringsSep " " ([
    "-I${SDL}/include/SDL"
    "-I${SDL_image}/include/SDL"
    "-I${SDL_ttf}/include/SDL"
  ]
  ++ stdenv.lib.optionals (stdenv.hostPlatform.isLinux) [
    "-I${gtk2.dev}/include/gtk-2.0"
    "-I${glib.dev}/include/glib-2.0"
  ]);
```

fgaz replied:
We can try when the darwin builder comes back up

fgaz commented:
Added hvl2wav and cleaned up some stuff. Like with lgpt, I propose to leave the cflags as is (it did build iirc) for now and try to split them later when the builder comes back up

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/319

thiagokokada commented:
Result of `nixpkgs-review pr 95797` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hivelytracker</li>
  </ul>
</details>

thiagokokada commented on pkgs/applications/audio/hivelytracker/default.nix:
```diff
@@ -0,0 +1,85 @@
+      its own instrument and module formats.
+    '';
+    license = licenses.bsd3;
+    platforms = platforms.all;
```
Seems like this is Linux only? So I would use `platforms.linux`.

fgaz replied:
It did build on darwin actually (when ofborg was still up)

fgaz replied:
I'll leave a todo in #81815 to check whether this is still true when hydra will build the package if that's ok

thiagokokada replied:
Are you sure? Asking this because of this: https://github.com/NixOS/nixpkgs/pull/95797/files#diff-0cfddbf69581aafdc210af5518f6fcc7f15b0926db12a2b2ffe56cd2a9d49f2aR38

thiagokokada replied:
BTW, AFAIK ofborg evaluates the derivation, not necessary builds them (this is the work of hydra, done only after the package is merged).

thiagokokada replied:
If `Makefile.linux` is really used to build macOS version too, I would add a comment, because I found this very confusing.

fgaz replied:
I got ahold of a darwin environment, rebased and tested this. It does not build. Marked as broken.

thiagokokada replied:
I still would set `platforms.linux` here (or `platforms.unix` if it is everything minus macOS), and after the package is fixed I would use `platforms.linux + platforms.darwin`.

fgaz replied:
I think broken is better in this case, since according to the manual:
* platforms is all platforms where the package _should_ work ("is supported")
* broken is all platform where the package should work but doesn't
and hydra won't try to build it anyway if it's marked as broken

SuperSandro2000 commented on pkgs/applications/audio/hivelytracker/default.nix:
```diff
@@ -0,0 +1,86 @@
+  ];
+
+  # Also build the hvl2wav tool
+  postBuild = "make -C hvl2wav";
```
```suggestion
  postBuild = ''
    make -C hvl2wav
  '';
```

fgaz replied:
Why the longer form?

SuperSandro2000 replied:
Commands are usually written like this to make it easier to add or remove them. Also it makes sure there is a final new line so that the command gets executed correct.

fgaz replied:
Didn't think about the newline. Fixed.

SuperSandro2000 commented on pkgs/applications/audio/hivelytracker/default.nix:
```diff
@@ -0,0 +1,86 @@
+    install -Dm755 hvl2wav/hvl2wav $out/bin/hvl2wav
+  '';
+
+  meta = with stdenv.lib; {
```
```suggestion
  meta = with lib; {
```

fgaz replied:
stdenv is already an input, while lib is not

OPNA2608 replied:
[doc: add "prefer lib over stdenv.lib" convention ](https://github.com/NixOS/nixpkgs/commit/c06b2b3d671da4847f950e7e9041920b40eca0bf)
[Deprecate use of stdenv.lib across Nixpkgs](https://github.com/NixOS/nixpkgs/issues/108938)

fgaz replied:
Ooh, that's nice! `stdenv.lib` always bothered me. Fixed.

SuperSandro2000 commented on pkgs/applications/audio/hivelytracker/default.nix:
```diff
@@ -0,0 +1,89 @@
+  ];
+
+  # Also build the hvl2wav tool
+  postBuild = ''
```
Can't we add that to makeFlags?

fgaz replied:
Nope, make doesn't support multiple `-C` flags like that. Unless we can do multiple make invocations within a single makeFlags

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 95797` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>hivelytracker</li>
</ul>
</details>
